onnxruntime/python/onnxruntime_inference_collection.py

```diff
@@ -346,6 +346,11 @@ def _create_inference_session(self, providers, provider_options, disabled_optimi
                                                                         provider_options,
                                                                         available_providers)
 
+        if providers == [] and 'CUDAExecutionProvider' in available_providers:
```

Comment:
I think we'll want to generalize this for all providers. if other execution providers are enabled in this ort build, then we can output the deprecation warning that providers can't be empty.

Reply:
Made it generalize.